app/models/parser/tiktok_item.rb

```diff
@@ -22,16 +23,27 @@ def parse_data_for_parser(doc, _original_url, _jsonld_array)
 
       handle_exceptions(StandardError) do
         @parsed_data[:raw][:api] = @parsed_data[:raw][:oembed] = get_tiktok_api_data(url)
-        match = url.match(TIKTOK_ITEM_URL)
+        if url.match(TIKTOK_ITEM_URL)
+          match = url.match(TIKTOK_ITEM_URL)
+          username = match['username']
+          external_id = match['id']
+          tag = ''
+        elsif url.match(TIKTOK_TAG_URL)  
+          username = ''
+          external_id = ''
+          tag = url.match(TIKTOK_TAG_URL)['tag']
+        end
+
         @parsed_data.merge!({
-          username: match['username'],
-          external_id: match['id'],
+          username: username,
+          external_id: external_id,
```

Comment:
`username` , `external_id` or `tag`  will be undefined in case there is no match for `TIKTOK_ITEM_URL` or `TIKTOK_ITEM_URL` so I suggest to define `username`, `external_id ` and `tag` with empty strings outside the if..else block (i.e. before line 26) and fill the right values inside `if ... elseif ... end` block

Reply:
I added an 'else' in case there is no match and tested with an url. What do you think?